controllers/AuthController.php

```diff
@@ -208,4 +220,22 @@ public function actionResetPassword($token)
             'model' => $model,
         ]);
     }
+
+    public function actionVerifyEmail($token)
+    {
+        $user = User::findByEmailVerificationToken($token);
+        if ($user) {
+            $user->email_verified = true;
+            $user->removeEmailVerificationToken();
+            if ($user->save()) {
```

Comment:
how about making this a method in `User` which can not fail on validation of unrelated fields?

`$user->verifyEmail();` will set email_verified, remove the token and save(false);

Reply:
I don't think it's a good idea to send an email from the User model. It's not model responsibility.